llvm/test/Transforms/SROA/atomic-vector.ll

```diff
@@ -0,0 +1,19 @@
+; RUN: opt < %s -passes='sroa' -S 2>&1 | FileCheck %s --check-prefix=ERR
+; RUN: opt < %s -passes='sroa' -S | FileCheck %s
+
+define float @atomic_vector() {
+; ERR-NOT: atomic load operand must have integer, pointer, or floating point type!
+; ERR-NOT:   <1 x float>  {{%.*}} = load atomic volatile <1 x float>, ptr {{%.*}} acquire, align 4
```

Comment:
It might not be necessary to check the error. It is sufficient to just check IR sequence. You will probably want to auto generate the check lines as well, which could make others life much easier in the future.

Reply:
Hm, right, since the IR sequence will either match or not. However, the error is illustrative of the actual issue, so I'm not sure if we should remove it.

Reply:
but that could prevent you from auto generating check lines. I'm currently modifying 50+ test cases manually and it is really annoying.

Reply:
besides, the error can be easily tracked by checking the git info and tracking down to this PR

Reply:
Thanks. Makes sense.